modules/hostap/src/supp_events.h

```diff
@@ -41,7 +42,99 @@ enum net_event_wpa_supp_cmd {
 #define NET_EVENT_WPA_SUPP_IFACE_REMOVING					\
 	(_NET_MGMT_WPA_SUPP_EVENT | NET_EVENT_WPA_SUPP_CMD_IFACE_REMOVING)
 
-int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int status);
+int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, void *status,
+			size_t len);
 int generate_supp_state_event(const char *ifname, enum net_event_wpa_supp_cmd event, int status);
 
+#define REASON_CODE_LEN 32
```

Comment:
Currently, `NO_PSK_AVAILABLE` is the longest string, so, should we just limit to 17? We also need to add a note here to update this if we add a new call to `wpas_auth_failed`

Reply:
or just add a enum here and handle the string to enum converion in an helper, this way we abstract the internal implementation.

Reply:
Since we are reading reason code as string from event, I think adding enum to string translation is not required here. Will reduce the array size to 18 bytes.